providers/gitlab.go

```diff
@@ -34,28 +39,89 @@ func NewGitLabProvider(p *ProviderData) *GitLabProvider {
 		p.ValidateURL = &url.URL{
 			Scheme: "https",
 			Host:   "gitlab.com",
-			Path:   "/api/v4/user",
+			Path:   "/api/v4",
 		}
 	}
-	if p.Scope == "" {
-		p.Scope = "read_user"
-	}
+	p.Scope += " read_user"
 	return &GitLabProvider{ProviderData: p}
 }
 
 // GetEmailAddress returns the Account email address
 func (p *GitLabProvider) GetEmailAddress(s *SessionState) (string, error) {
+	if p.Group != "" {
+		groupsJSON, groupsJSONError := data(p.ValidateURL.String()+"/groups?access_token="+s.AccessToken, s)
+		if groupsJSONError != nil {
+			return "", groupsJSONError
+		}
+		groupsValid := false
+		for i := range groupsJSON.MustArray() {
+			group, groupJSONError := groupsJSON.GetIndex(i).Get("name").String()
+			if groupJSONError == nil {
+				groupsValid = groupsValid || (strings.ToLower(group) == p.Group)
+			}
+		}
+		if !groupsValid {
+			return "", fmt.Errorf("user has no acces to group %s", p.Group)
+		}
+	}
+
+	singlemailJSON, singlemailJSONError := data(p.ValidateURL.String()+"/user?access_token="+s.AccessToken, s)
```

Comment:
I think these variable names could be improved for readability. Perhaps something like `userEmailJSON` and `userEmailJSONError` would give an instant a hint to their purpose? Also, they should be in camelCase.

Reply:
Thanks for reviewing! I picked some more explicit names.